plugins/cccongress.py

```diff
@@ -0,0 +1,280 @@
+                """Save hall information in event and add to announcelist"""
+                event['hall'] = hall
+                announcelist.append(event)
+
+    """Announce found events"""
+    if len(announcelist) > 0:
+        bot.privmsg(
+            bot.config.autojoins[0],
+            "The following talks will start in %d minutes" % (
+                config['announce_minutes']
+            )
+        )
+        for event in announcelist:
+            yield from _output_single_talk(event['hall'],
+                                           event,
+                                           bot,
+                                           bot.config.autojoins[0])
+
+
+@asyncio.coroutine
+def _output_talks(slot, bot, target):
+    """Output talks
+
+        slot: Output talks running now (slot = 0) or next (slot = 1)
+    """
+
+    event_counter = 0
+    for hall in _get_halls():
+        event = _get_talk(hall, slot)
+        if event is not None:
+            event_counter += 1
+            yield from _output_single_talk(hall, event, bot, target)
+
+    if event_counter == 0:
+        bot.privmsg(target, "No talks found.")
+
+
+@asyncio.coroutine
+def _output_single_talk(hall, event, bot, target):
+    """Output single talk
+
+        hall: Name of hall
+        event: Event information
+    """
+
+    bot.privmsg(
+        target,
+        "[%s] %s %s" % (hall, event['start'], event['title'])
+    )
+    bot.privmsg(
+        target,
+        "%s (%s / %s)" % (
+            '{:<14}'.format(''), event['language'], _get_persons(event)
+        )
+    )
+
+
+@asyncio.coroutine
+def _schedule_information(bot, target):
+    """Output information about schedule"""
+
+    config = BYTEBOT_PLUGIN_CONFIG['cccongress']
+    json_data = _get_json_data()
+
+    bot.privmsg(
+        target,
+        "%s (%s)" % (
+            json_data['schedule']['conference']['title'],
+            json_data['schedule']['conference']['acronym']
+        )
+    )
+
+    file_mtime = os.path.getmtime(config['cache'])
+    file_dtime = datetime.datetime.fromtimestamp(file_mtime)
+    bot.privmsg(
+        target,
+        "Schedule was last updated on %s" % (
+            file_dtime.strftime('%Y-%m-%d %H:%M:%S')
+        )
+    )
+
+    bot.privmsg(
+        target,
+        "Current schedule version is %s" % json_data['schedule']['version']
+    )
+
+
+@asyncio.coroutine
+def _output_help(bot, target):
+    """Output help"""
+
+    bot.privmsg(target, "The following commands are available:")
+    bot.privmsg(target, "!cccongress          - Show current talks")
+    bot.privmsg(target, "!cccongress halls    - Show available halls")
+    bot.privmsg(target, "!cccongress next     - Show next talks")
+    bot.privmsg(target, "!cccongress schedule - Show schedule information")
+
+
+def _get_json_data():
+    """Get json from cached file"""
+
+    with open(BYTEBOT_PLUGIN_CONFIG['cccongress']['cache']) as json_file:
+        return json.load(json_file)
```

Comment:
Good catch, fixed Exception handling in many places - also http update exception handling